### PR TITLE
[RAPTOR-9455] bump scikit-learn==1.0.2, release 1.10.5

### PR DESCRIPTION
**Pull request by yakov-g**
# This repository is public. Do not put here any private DataRobot or customer's data: code, datasets, model artifacts, .etc.

## Summary
* bump scikit-learn==1.0.2 in public envs
* bump datarobot-mlops==9.0.7 in tests
* replace deprecated df.append() with pandas.concat()

### CODE REVIEW

**Comment by yakov-g** [comment]
jenkins run test_integration_r_environment

**Comment by yakov-g** [comment]
jenkins run test_integration_r_environment

**Comment by yakov-g** [comment]
jenkins run jenkins run

**Comment by yakov-g** [comment]
jenkins run test_integration_general

**Comment by yakov-g** [comment]
jenkins run test_integration_general

**Comment by yakov-g** [comment]
jenkins run test_integration_general

**Comment by yakov-g** [comment]
jenkins run test_integration_general

**Comment by yakov-g** [comment]
jenkins run test_integration_python_sklearn

**Comment by yakov-g** [comment]
jenkins run test_integration_python_sklearn

**Comment by yakov-g** [comment]
jenkins run test_integration_r_environment

**Comment by yakov-g** [comment]
jenkins run test_integration_general

**Review by zohar-mizrahi — APPROVED** [review]

File: public_dropin_environments/python3_keras/requirements.txt [hunk 1]
```diff
@@ -1,6 +1,6 @@
 tensorflow==2.11.1
 numpy==1.22.0
 pandas<=1.5.3
-scikit-learn==0.24.2
+scikit-learn==1.0.2
```
Comment (akshoop) [inline public_dropin_environments/python3_keras/requirements.txt:4]: CC @eric-s-s @scottp-ml for awareness of `scikit-learn` version bump to `1.0.2` for public dropin envs

File: requirements_test.txt [hunk 2]
```diff
@@ -10,5 +10,7 @@ PyYAML>=3.11
 requests >= 2.24.0
 responses
 retry
-scikit-learn==0.24.2
+# Note: keep it as this file is installed in quantum, which is Py3.7
+scikit-learn==1.0.2; python_version=="3.7"
+scikit-learn==1.0.2; python_version>="3.8"
```
Comment (akshoop) [inline requirements_test.txt:15]: These are the same version. Did you instead mean to do the following?
```suggestion
scikit-learn==0.24.2; python_version=="3.7"
scikit-learn==1.0.2; python_version>="3.8"
```

**Review by akshoop — APPROVED** [review]
LGTM from Core Modeling with minor comment

File: requirements_test.txt (hunk 2 above)
Comment (yakov-g) [reply requirements_test.txt:15]: No,
I meant that once we bump scikit-learn to 1.1.X
we need to keep
scikit-learn==1.0.2; python_version=="3.7"

**Pull request merged**